stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TextDecorator.kt

```diff
@@ -0,0 +1,80 @@
+package io.getstream.chat.android.ui.messages.adapter.viewholder.decorator
+
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import com.getstream.sdk.chat.adapter.MessageListItem
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.messages.adapter.viewholder.GiphyViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.MessagePlainTextViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyFileAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithFileAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.view.MessageListItemStyle
+
+internal class TextDecorator(val style: MessageListItemStyle) : BaseDecorator() {
+
+    private fun decorateMessageText(
+        textView: TextView,
+        isMine: Boolean,
+    ) {
+        val defaultTextColor = ContextCompat.getColor(textView.context, R.color.stream_ui_black)
+
+        val defaultLinkTextColor = ContextCompat.getColor(textView.context, R.color.stream_ui_accent_blue)
+
+        val textColorMine = style.messageTextColorMine ?: defaultTextColor
+
+        val textColorTheirs = style.messageTextColorTheirs ?: defaultTextColor
+
+        val linkColorTextMine = style.messageLinkTextColorMine ?: defaultLinkTextColor
+
+        val linkColorTextTheirs = style.messageLinkTextColorTheirs ?: defaultLinkTextColor
+
+        val textColor = if (isMine) textColorMine else textColorTheirs
+
+        val linkColor = if (isMine) linkColorTextMine else linkColorTextTheirs
```

Comment:
We compute more values here than we'll use, I'd go with

```suggestion
        val textColor = (if (isMine) style.messageTextColorMine else style.messageTextColorTheirs) 
            ?: ContextCompat.getColor(textView.context, R.color.stream_ui_black)
        val linkColor = (if (isMine) style.messageLinkTextColorMine else style.messageLinkTextColorTheirs)
            ?: ContextCompat.getColor(textView.context, R.color.stream_ui_accent_blue)
```

Reply:
I was attempting to make it easy to read, but I see your point. I'll see what I can do.